mtags/src/main/scala/scala/meta/internal/mtags/JavaMtags.scala

```diff
@@ -0,0 +1,172 @@
+package scala.meta.internal.mtags
+
+import com.thoughtworks.qdox._
+import com.thoughtworks.qdox.model.JavaClass
+import com.thoughtworks.qdox.model.JavaField
+import com.thoughtworks.qdox.model.JavaMember
+import com.thoughtworks.qdox.model.JavaMethod
+import com.thoughtworks.qdox.model.JavaModel
+import com.thoughtworks.qdox.parser.ParseException
+import java.io.StringReader
+import java.util.Comparator
+import scala.collection.mutable
+import scala.meta.inputs.Input
+import scala.meta.inputs.Position
+import scala.meta.internal.semanticdb.Language
+import scala.meta.internal.semanticdb.SymbolInformation.Kind
+import scala.meta.internal.semanticdb.SymbolInformation.Property
+import scala.meta.internal.mtags.Enrichments._
+
+/**
+ * Utility to generate method symbol  disambiguators according to SemanticDB spec.
+ *
+ * See https://scalameta.org/docs/semanticdb/specification.html#scala-symbol
+ */
+final class OverloadDisambiguator(
+    names: mutable.Map[String, Int] = mutable.Map.empty
+) {
+  def disambiguator(name: String): String = {
+    val n = names.getOrElseUpdate(name, 0)
+    names(name) = n + 1
+    if (n == 0) "()"
+    else s"(+$n)"
+  }
+}
+
+object JavaMtags {
+  private implicit class XtensionJavaModel(val m: JavaModel) extends AnyVal {
+    def lineNumber: Int = m.getLineNumber - 1
+  }
+
+  def index(input: Input.VirtualFile): MtagsIndexer = {
+    val builder = new JavaProjectBuilder()
+    new MtagsIndexer { self =>
+      override def language: Language = Language.JAVA
+      override def indexRoot(): Unit = {
+        try {
+          val source = builder.addSource(new StringReader(input.value))
+          if (source.getPackage != null) {
+            source.getPackageName.split("\\.").foreach { p =>
+              pkg(
+                p,
+                toRangePosition(source.getPackage.lineNumber, p)
+              )
+            }
+          }
+          source.getClasses.forEach(visitClass)
+        } catch {
+          case _: ParseException =>
+          case _: NullPointerException => ()
+          // Hitting on this fellow here indexing the JDK
+          // Error indexing file:///Library/Java/JavaVirtualMachines/jdk1.8.0_102.jdk/Contents/Home/src.zip/java/time/temporal/IsoFields.java
+          // java.lang.NullPointerException: null
+          // at com.thoughtworks.qdox.builder.impl.ModelBuilder.createTypeVariable(ModelBuilder.java:503)
+          // at com.thoughtworks.qdox.builder.impl.ModelBuilder.endMethod(ModelBuilder.java:470)
+          // TODO(olafur) report bug to qdox.
```

Comment:
Has this been reported? 😇

Reply:
I don't think so, didn't manage to minimize the error.  Replaced the comment with explanation why we ignore parse errors.